h2o-r/h2o-package/R/targetencoding.R

```diff
@@ -0,0 +1,104 @@
+# This file is auto-generated by h2o-3/h2o-bindings/bin/gen_R.py
+# Copyright 2016 H2O.ai;  Apache License Version 2.0 (see LICENSE for details) 
+#'
+# -------------------------- Target Encoding -------------------------- #
+#' 
+#' Creates a target encoding frame based on a group-by column and a target column (numeric or binary).
+#' 
+#' Calculates the mean of the target column per group removing the value of the existing row.
+#' This can help predictive performance of high cardinality categorical columns in supervised learning problems.
+#' 
+#' @param group_column An H2OFrame object with a single categorical column.
+#' @param y An H2OFrame object with a single numeric column.
+#' @param domain (Optional) Vector with response factor for classification.
+#' @param noise_level (Optional) The amount of random noise added to the target encoding.  This helps prevent overfitting. Defaults to 0.01 * range of y.
+#' @param seed (Optional) A random seed used to generate draws from the uniform distribution for random noise. Defaults to -1.
+#' @return Returns an H2OFrame object containing the leave one out target encoding.
+#' @examples
+#' \donttest{
+#' library(h2o)
+#' h2o.init()
+#' 
+#' # Run Target Encoding on bank-additional-full data with numeric `y`
+#' data.hex = h2o.importFile(
+#' path = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/demos/bank-additional-full.csv",
+#' destination_frame = "data.hex")
+#' h2o.target_encoding(data.hex$job, data.hex$age)
+#' 
+#' # Run Target Encoding on bank-additional-full data with binary `y`
+#' h2o.target_encoding(data.hex$job, data.hex$y)
+#' }
+#' @export
+
+h2o.target_encoding <- function(group_column, y, domain = NULL, noise_level = NULL, seed = -1){
```

Comment:
The purpose of the domain parameter is to map levels to indices, is that correct?

Perhaps we can reflect it better in the API so that the purpose is manifested in the function's signature.